### DISCUSSION TITLE
Device broke after flashing new firmware

### DISCUSSION THREAD

**Discussion by petersohn** [arendst/Tasmota#16867] [Q&A] [answered]
I have a [Delock 11827](https://templates.blakadder.com/delock_11827.html) device, which is preinstalled with Tasmota. I could configure the wifi fine and log in to the web interface. However, it is in German. I wanted to change it to an English firmware, but I couldn't find a way with its official firmware. So I clicked Firmware Upgrade, and uploaded the official `tasmota-minimal.bin` (downloaded from Github) file (because the full `tasmota.bin` failed). After that, the device restarted, and it doesn't work. It doesn't connect to my wifi network and doesn't enter AP mode either. What can I do in this case? Is there anything I can do apart from taking it apart and reflashing the firmware through serial port?

**Comment by barbudor** [comment, marked as answer]
If you uploaded minimal from a Tasmota that was properly configured and connected to your wifi, it should have reconnected to your wifi
Do you remember doing anything else?

Are you 100% sure that it is not connected possibly with another ip address? Did you ran an ip scanner?

**Reply by petersohn** [reply]
It doesn't appear on my router's client list, and I cannot connect to the old IP address.

What I did was I tried to flash a full firmware, which failed, and then I flashed the minimal one. Could it have caused the problem?

**Reply by Jason2866** [reply]
Flashing the a minimal version which does not match to the installed version does end in a bricked device.

**Reply by barbudor** [reply]
It should have not
Actually what you did is the process to go from one big firmware to another big firmware with minimal in the middle
May be it could have been simpler to use the http download mode (the top part of the upgrade windows) by pointing to http://ota.tasmota.com/tasmota/tasmota.bin.gz
But manually uploading minimal should have worked.
Minimal doesn't allow to make change to settings but if settings were already there, it should have reconnected to your wifi.

Just as a last resort you may try to reset settings using one of the recovery method:
- press 40+ sec on the button and let restart.
- or put the plug on a power strip with a switch on it and power cycle on->off and off->on 7 times with keeping around 3~4 sec in each state

Tasmota should reboot in AP mode
The problem is that with minimal you won't be able to set new wifi creds and you have 3 minutes to go to 192.168.4.1/up to upgrade to your firmware

If that doesn't work, I'm afraid you have to Crack it open

**Reply by sfromis** [reply]
It is possible that Delock created their firmware build in a way not directly compatible with regular Tasmota, having a different code (`CFG_HOLDER`) causing a full reset when the new binary booted. And if that binary was -minimal, that means bricked device. While I don't know if this was the case, it is consistent with your symptoms.

**Reply by sfromis** [reply]
Their product page for [Delock 11827](https://www.delock.com/produkt/11827/merkmale.html) only mentions Tasmota versions up to 7.2, meaning that a [stepwise upgrade](https://tasmota.github.io/docs/Upgrading/#upgrade-flow) to current Tasmota versions would be required.

**Reply by petersohn** [reply]
> If that doesn't work, I'm afraid you have to Crack it open

None of the recovery methods worked, so I'd just need to do that. And by that, I mean literally crack it open, because this damn thing doesn't even have screws.

**Reply by barbudor** [reply]
I did assume that you were doing an language change between 2 versions that were same or close
But indeed as pointed by Jason and Sfromis, if the original version was outdated by further than more than 2 major version then it's bricked

Opening a glued device is challenging but possible although with a risk of breaking it

**Reply by petersohn** [reply]
I managed to physically open the device, but I cannot flash it because I could not reach the necessary ports. Besides, IO ports 0 and 2 are connected, which might also cause problems.

Anyway, on another device I followed the proper upgrade procedure and I could upgrade it without any problems.
